src/gcode-machine-control.cc

```diff
@@ -530,13 +529,21 @@ void GCodeMachineControl::Impl::set_output_flags(HardwareMapping::LogicOutput ou
 }
 
 void GCodeMachineControl::Impl::get_current_position() {
-  if (planning_buffer_.size() > 0) {
-    const struct AxisTarget *current = planning_buffer_[0];
-    const int *mpos = current->position_steps;
-    const float x = 1.0f * mpos[AXIS_X] / cfg_.steps_per_mm[AXIS_X];
-    const float y = 1.0f * mpos[AXIS_Y] / cfg_.steps_per_mm[AXIS_Y];
-    const float z = 1.0f * mpos[AXIS_Z] / cfg_.steps_per_mm[AXIS_Z];
-    const float e = 1.0f * mpos[AXIS_E] / cfg_.steps_per_mm[AXIS_E];
+    int i, motor;
+    const MotorCoordSystem
+        &motor_position_steps_ = motor_ops_->GetCurrentMotorPos();
+    FixedArray<int,GCODE_NUM_AXES> axis_position_steps_;
```

Comment:
Same thing here: no underscore in local variables (ok, I'll not mention it again).